contributor-docs/adrs/adr-xxx-behavior-isolation.md

```diff
@@ -0,0 +1,137 @@
+
+React's core library does not offer a way to represent React's component tree in either the real DOM, nor on the server. In this way, developers can use either the `react-dom` library on the client, or the `react-dom/server` library using a runtime like NodeJS on the server to deliver a similar experience to traditional server side applications built in frameworks like Ruby on Rails. This is sometimes referred to as "isomorphic" code. The unique selling point of "isomorphic" components, is that developers can write a component once and have it execute in both environments, delivering a string of HTML in a server response, and reconciling DOM operations on the client.
+
+ADR 002 describes complexity around Custom Elements and Server Side Rendering. It claims that for Custom Elements to be well supported, that Primer React would need to accumulate added complexity which would be surfaced to the user, going on to claim that while libraries may exist their usage is antithetical to Custom Element patterns.
+
+Primer React's [recent addition of the Relative Time component](https://github.com/primer/react/pull/2484) (itself a wrapper of the [`relative-time`](https://github.com/github/relative-time-element) custom element) demonstrates that a custom element can be loaded into a server side runtime such as NodeJS without causing error, by adding [2 additional lines of code to the element itself](https://github.com/github/relative-time-element/blob/4ab3f2807a02dfd687b2b7fa04553300ff87ddbe/src/relative-time-element.ts#L2-L3):
+
+```typescript
+const root = (typeof globalThis !== 'undefined' ? globalThis : window) as typeof window
+const HTMLElement = root.HTMLElement || (null as unknown as typeof window['HTMLElement'])
+```
+
+These lines could also be added to the global context, to avoid adding it to each element.
+
+The `RelativeTime` react component uses `@lit-labs/react`, which makes integrating the component into React painless, and requires [1 line of code for integration](https://github.com/primer/react/blob/d995bb8485fa1dfd0a3637b4851c5ac539b8416d/src/RelativeTime/RelativeTime.tsx#L8):
+
+```typescript
+const RelativeTime = styled(createComponent(React, 'relative-time', RelativeTimeElement))(sx)
+```
+
+There are no additional steps required for consumers of Primer React, and the component behaves as if it is a first-class React component, supporting the same properties and events as any regular React component. The server will render the Custom Element tag (in the case of the `RelativeTime` component, `<relative-time>`) and the client is able to assign class properties and use callbacks for events, as expected.
```

Comment:
Some of our main goals with SSR which may difficult to achieve with custom elements:

* Rendering without JavaScript
* Reducing CLS

`RelativeTime` is a perfect example of a component which we _could_ render with SSR, but will be unable to if it is wrapped in a custom element. By performing client-only rendering, we miss out on both of these SSR goals. If we instead had a `relativeTime` vanilla JS behavior (with a callback when it needed to update), we could write a very small wrapper in React which would function in both SSR and CSR environments. Additionally, that JS behavior could be consumed by the `<relative-time>` custom element for use within PVC.